### PR TITLE
Predicate refactor

### PR DESCRIPTION
**Pull request by bobismijnnaam**
Improve support for predicates.

- Inline predicates can be scaled.
- Add tests for inline predicates.
- Unfolding and folding of inline predicates is now encoded as asserting the predicate body.

### CODE REVIEW

File: src/main/java/vct/col/rewrite/InlinePredicatesRewriter.java [hunk 1]
```diff
@@ -68,43 +84,97 @@ public void visit(Method m){
...
         break;
     }
   }
+
+  @Override
+  public void visit(ASTSpecial e) {
+      switch (e.kind) {
+        case Fold:
+        case Unfold:
+          if (inline(e.getArg(0))) {
```
Comment (niomaster) [inline src/main/java/vct/col/rewrite/InlinePredicatesRewriter.java:169]: Still think we should discuss this in the next meeting :)

**Review by niomaster — APPROVED** [review]

File: src/main/java/vct/col/rewrite/InlinePredicatesRewriter.java (hunk 1 above)
Comment (bobismijnnaam) [reply src/main/java/vct/col/rewrite/InlinePredicatesRewriter.java:169]: Yes it is in my notes! ;)

**Comment by niomaster** [comment]
The decision from the meeting was to consider unfolding or folding an inline predicate a no-op, but warn that the statement is meaningless.

**Comment by raulmonti** [comment]
@bobismijnnaam could you look into this PR and update it?

**Comment by bobismijnnaam** [comment]
Continued at #639.

**Pull request closed without merging**